src/databricks/labs/ucx/install.py

```diff
@@ -112,68 +110,63 @@ def extract_major_minor(version_string):
     return None
 
 
-class WorkspaceInstaller:
+class WorkspaceInstaller(WorkspaceContext):
     def __init__(
         self,
-        prompts: Prompts,
-        installation: Installation,
         ws: WorkspaceClient,
-        product_info: ProductInfo,
         environ: dict[str, str] | None = None,
         tasks: list[Task] | None = None,
     ):
+        super().__init__(ws)
         if not environ:
             environ = dict(os.environ.items())
+        self._force_install = environ.get("UCX_FORCE_INSTALL")
         if "DATABRICKS_RUNTIME_VERSION" in environ:
             msg = "WorkspaceInstaller is not supposed to be executed in Databricks Runtime"
             raise SystemExit(msg)
-        self._ws = ws
-        self._installation = installation
-        self._prompts = prompts
-        self._policy_installer = ClusterPolicyInstaller(installation, ws, prompts)
-        self._product_info = product_info
-        self._force_install = environ.get("UCX_FORCE_INSTALL")
-        self._is_account_install = environ.get("UCX_FORCE_INSTALL") == "account"
+        try:
+            installation = self.product_info.current_installation(ws)
+        except NotFound:
+            if self._force_install == "user":
+                installation = Installation.assume_user_home(ws, self.product_info.product_name())
+            else:
+                installation = Installation.assume_global(ws, self.product_info.product_name())
+        self.replace(installation=installation)
+
+        self._is_account_install = self._force_install == "account"
         self._tasks = tasks if tasks else Workflows.all().tasks()
 
     def run(
         self,
         default_config: WorkspaceConfig | None = None,
         verify_timeout=timedelta(minutes=2),
-        sql_backend_factory: Callable[[WorkspaceConfig], SqlBackend] | None = None,
-        wheel_builder_factory: Callable[[], WheelsV2] | None = None,
         config: WorkspaceConfig | None = None,
     ) -> WorkspaceConfig:
-        logger.info(f"Installing UCX v{self._product_info.version()}")
+        logger.info(f"Installing UCX v{self.product_info.version()}")
         if config is None:
             config = self.configure(default_config)
-        if not sql_backend_factory:
-            sql_backend_factory = self._new_sql_backend
-        if not wheel_builder_factory:
-            wheel_builder_factory = self._new_wheel_builder
-        wheels = wheel_builder_factory()
-        install_state = InstallState.from_installation(self._installation)
+        install_state = InstallState.from_installation(self.installation)
```

Comment:
we already have the `self.install_state` from `GlobalContext`

Reply:
done